### ISSUE TITLE
Cannot perform fit without fHdNdEpBkg 

### ISSUE THREAD

**Issue by StFroese** [javierrico/gLike#67]
I get the following error message, when I'm trying to use jointLklDM with a root file generated from glikeInputs.
Seems like glikeInputs doesn't write fHdNdEpBkg to the root file but according to the gLike wiki fHdNdEpBkg is not mandatory for the fit.

```
****************************************************************
*** SUMMARY OF LKL TERMS: 
****************************************************************
 *              Object Name : JointLkl_00
 *      Error in Units of G : 0
 *             # of samples = 1
 ****************************************
                                        *              Object Name : Iact1dUnbinnedLkl_01
                                        *      Error in Units of G : 0
                                        *                E' range  = [50,1e+06] GeV
                                        *                 Non/Noff = 11665/10192
                                        *             Measured tau = 0.873715 +/- 0.00568982
                                        *         Observation time = 31.6523 h
                                        *                   log(J) = (19.42 +/- 0) GeV^2 cm^-5  or  GeV cm^-2
                                        *               Aeff histo : YES
                                        *           Off Aeff histo : NO
                                        *       E Reso/Bias graphs : YES
                                        *         Migration matrix : YES
                                        *    Background rate histo : NO
                                        *    Foreground rate histo : NO
                                        *       Signal dN/dE histo : NO
                                        *      Signal dN/dE' histo : NO
                                        *  Signal dN/dE' Off histo : NO
                                        ****************************************

***********************************************************************************
**** LOOPING OVER MASSES, CONFIGURE MASS-DEPENDENT HISTOS AND CALL LKL MINIMIZATION
****
**** DM mass = 100 GeV  (1/10)
**************************************
 *** Read or compute dN/dE and dN/dE' histos for each samples:
  ** Building signal histos for sample Iact1dUnbinnedLkl_01:
   * Reading dN/dE for signal from file .//DM/dNdE/Cirelli//dNdESignal_bb_100.0mass.root with BR = 1... Ok!
Iact1dUnbinnedLkl::CheckHistograms Message: will create fHdNdEpSignal from fHdNdESignal, fHAeff & fMigMatrix... Done! 
 *** End of reading dN_signal/dE and dN_signal/dE' histograms
Iact1dUnbinnedLkl::CheckHistograms Warning: fHdNdEpBkg histogram missing!!
Iact1dUnbinnedLkl::MakeChecks Warning: missing information, cannot perform fit, check your code!

 *** Break *** segmentation violation
```

MARS-Version: V2-20-1
gLike-Version:v00.10.00

**Comment by javierrico** [comment]
Dear @StFroese 
The background model is needed when doing the unbinned likelihood (i.e using a Iact1dUnbinnedLkl instance)
If you do not have it then you can still use the binned likelihood (change Iact1dUnbinnedLkl by Iact1dBinnedLkl in the rcfile and configure it accordingly).
Let me know if that works or if you have any other question
Cheers

**Comment by StFroese** [comment]
I've tried it. I'm getting a different error now: 
```
***********************************************************************************
**** LOOPING OVER MASSES, CONFIGURE MASS-DEPENDENT HISTOS AND CALL LKL MINIMIZATION
****
**** DM mass = 100 GeV  (1/10)
**************************************
 *** Read or compute dN/dE and dN/dE' histos for each samples:
  ** Building signal histos for sample Iact1dBinnedLkl_01:
   * Reading dN/dE for signal from file .//DM/dNdE/Cirelli//dNdESignal_bb_100.0mass.root with BR = 1... Ok!
Iact1dUnbinnedLkl::CheckHistograms Message: will create fHdNdEpSignal from fHdNdESignal, fHAeff & fMigMatrix... Done! 
  ** Building signal histos for sample Iact1dBinnedLkl_02:
   * Reading dN/dE for signal from file .//DM/dNdE/Cirelli//dNdESignal_bb_100.0mass.root with BR = 1... Ok!
Iact1dUnbinnedLkl::CheckHistograms Message: will create fHdNdEpSignal from fHdNdESignal, fHAeff & fMigMatrix... Done! 
 *** End of reading dN_signal/dE and dN_signal/dE' histograms
Iact1dBinnedLkl::BuildAndBinOnOffHistos (Iact1dBinnedLkl_01) Message: fHNON/fHNOff original number of bins = 10, rebinned to 6 bins to keep minimum of 1 events per bin

 *** Break *** segmentation violation
```

**Comment by StFroese** [comment]
My joint likelihood ist defined like this:
```
jointLklDM.lklTerm000: JointLkl -   DlogJ=0
jointLklDM.lklTerm001: Iact1dBinnedLkl    0   logJ=19.42  DlogJ=0  inputfile=Output_glikeInputs_w0.root nbins=20
jointLklDM.lklTerm001: Iact1dBinnedLkl    0   logJ=19.42  DlogJ=0  inputfile=Output_glikeInputs_w1.root nbins=20
```

**Comment by TjarkMiener** [comment]
There is a typo in the class counter, @StFroese! Second jointLklDM.lklTerm001 should be jointLklDM.lklTerm00**2**. Hope this is solving the segmentation fault.

**Comment by StFroese** [comment]
Ah sry there is actually a 002 in the input card. The typo happened during copy&paste. The error remains.

**Comment by javierrico** [comment]
Hi @StFroese 
This might be an actual code bug, could you send me a (small as possible) set of rcfile+input file(s) producing this error so that I can try to reproduce, understand and fix it?
Thanks!

**Comment by StFroese** [comment]
@javierrico yeah sure. I will do so.

**Comment by javierrico** [comment]
Hi @StFroese 
I have pushed a bug fix to branch v00.10
I managed to run jointLklDM with your rcfile and input files
There are still some warnings due to low statistics that I will try to fix before releasing v00.10.01 bug fix, but in the meantime it would be very useful if you could try v00.10 and let me know if it also works for you
Thanks!

**Comment by StFroese** [comment]
@javierrico seems to work now. Thanks a lot!

**Comment by javierrico** [comment]
Fix xill be included in v00.10.01
